### DISCUSSION TITLE
Can 'node-ytdl-core' run in browser completely with little modification

### DISCUSSION THREAD

**Discussion by ashish-singh-kr** [fent/node-ytdl-core#858] [General]
Hi sir,

Can 'node-ytdl-core' run in browser completely with little modification. (For example I want to run the html downloader and parser in browser rather than at server, as youtube will block server ip address if it gets too many calls. )

TIA

**Comment by fent** [comment]
it can run in web extensions. you can bundle it for browser with rollup or browserify

**Reply by Haiz14** [reply]
I was able to get yt-dl working working on browser but there's some CORS issue

```log
Access to fetch at 'https://www.youtube.com/watch?v=awNaynJbJqI&hl=en' from origin 'http://localhost:5173' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

ny ideas on how to overcome it ?

**Reply by TasseDeCafe** [reply]
You need to set up CORS in your backend. If you're developing a browser extension, you'll need to use ngrok to route your requests to a HTTPS address. Let me know if you need help with that.

I am trying to get yt-dl working on browser too, how did you do it? I'd like to get access to the JSON with the video details from the browser.

**Reply by anonwins** [reply]
I'm interested in that too. I'd like a quick guide on how to:
a) use ytdl-core in browser
and b) solve the cors issue
Thanks
